lib/tlscontext.c

```diff
@@ -513,20 +516,84 @@ tls_context_setup_dh(TLSContext *self)
   return ctx_dh_success;
 }
 
+static gboolean
+tls_context_load_pkcs12(TLSContext *self)
+{
+  if (!file_exists(self->pkcs12_file))
+    return FALSE;
+
+  FILE *p12_file = fopen(self->pkcs12_file, "rb");
+
+  if (!p12_file)
+    return FALSE;
+
+  PKCS12 *pkcs12 = d2i_PKCS12_fp(p12_file, NULL);
+  fclose(p12_file);
+
+  if (!pkcs12)
+    return FALSE;
+
+  EVP_PKEY *private_key;
+  X509 *cert;
+  STACK_OF(X509) *ca_list = NULL;
+
+  if(!PKCS12_parse(pkcs12, "", &private_key, &cert, &ca_list))
+    {
+      PKCS12_free(pkcs12);
+      return FALSE;
+    }
+
+  PKCS12_free(pkcs12);
```

Comment:
What do you think about a `static gboolean _load_pkcs12_content(PKCS12 *pkcs12, EVP_PKEY **private_key, X509 **cert,STACK_OF(X509) **ca_list);` function?